File: SWEA/D2/1974. 스도쿠 검증/스도쿠 검증.py
def sudoku(array):
    for i in range(9):
        cnt = [0] * 10
        for j in range(9):
            if cnt[array[i][j]] == 0:
                cnt[array[i][j]] = 1
            else:
                return False
    
    for j in range(9):
        cnt = [0] * 10
        for i in range(9):
            if cnt[array[i][j]] == 0:
                cnt[array[i][j]] = 1
            else:
                return False
    
    for i in range(0, 9, 3):
        for j in range(0, 9, 3):
            cnt = [0] * 10
            for x in range(3):
                for y in range(3):
                    if cnt[array[x][y]] == 0:
                        cnt[array[x][y]] = 1
                    else:
                        return False
    
    return True
                    
t = int(input())
for test in range(1, t + 1):
    array = [list(map(int, input().split())) for _ in range(9)]
    result = sudoku(array)

    print(f"#{test} ", end='')
    if result:
        print(1)
    else:
        print(0)

File: SWEA/D2/1974. 스도쿠 검증/스도쿠 검증.java
import java.util.Scanner;

public class Solution {

	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(System.in);
		
		int t = sc.nextInt();
		for (int tc = 1; tc <= t; tc++) {
			int[][] sudoku = new int[9][9];
			for (int i = 0; i < 9; i++) {
				for (int j = 0; j < 9; j++) {
					sudoku[i][j] = sc.nextInt();
				}
			}
			
			int result = 1;
			for (int i = 0; i < 9; i++) {
				int[] cnt = new int[10]; 
				for (int j = 0; j < 9; j++) {
					if (cnt[sudoku[i][j]] == 0)
						cnt[sudoku[i][j]] = 1;
					else
						result = 0;
						
				}
			}
			
			for (int j = 0; j < 9; j++) {
				int[] cnt = new int[10]; 
				for (int i = 0; i < 9; i++) {
					if (cnt[sudoku[i][j]] == 0)
						cnt[sudoku[i][j]] = 1;
					else
						result = 0;
						
				}
			}
			
			for (int i = 0; i < 9; i += 3) {
				for (int j = 0; j < 9; j += 3) {
					int[] cnt = new int[10];
					for (int x = 0; x < 3; x++) {
						for (int y = 0; y < 3; y++) {
							if (cnt[sudoku[x][y]] == 0)
								cnt[sudoku[x][y]] = 1;
							else
								result = 0;
						}
					}
				}
			}
			
			System.out.printf("#%d %d%n", tc, result);
		}
	}

}